src/signal_scope/plotwidget.cpp

```diff
@@ -305,6 +343,12 @@ void PlotWidget::onResetYAxisScale()
   {
     area = QRectF(-1, -1, 2, 2);
   }
+  else
+  {
+    double margin = std::abs(area.top()*0.1);
```

Comment:
at this point in the code, i think the rectangle has a valid height, so perhaps the margin should be a percentage of the height, and not a percentage of the absolute coordinate of the top or bottom?